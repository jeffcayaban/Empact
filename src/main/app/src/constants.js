export const API_BASE_URL = 'http://localhost:5000/api';
export const ACCESS_TOKEN = 'accessToken';
export const CURRENT_USERNAME = 'username';

export const AUTHORIZATION_HEADER_FIELD = "Authorization";
export const AUTHORIZATION_TOKEN_TYPE = "Bearer";

// Website Name
export const WEBSITE_NAME = "Empact";

// User Roles
export const ROLE_USER = "ROLE_USER";
export const ROLE_ADMIN = "ROLE_ADMIN";

// Argument Types
export const EXPERT_OPINION = "ExpertOpinion";
export const POPULAR_OPINION = "PopularOpinion";

// Argumentation Schemes
export const ARGUMENTATION_SCHEMES = [
    'By Expert Opinion',
    'By Popular Opinion'
];

// Critical Question IDs
export const CRITICAL_QUESTION_IDS = [
    'CQ1',	// Are the circumstances true?
    'CQ2',
    'EO_CQ1',
    'PO_CQ1'
];

// Content Types
export const USER = "User";
export const PETITION = "Petition";
export const ARGUMENT = "Argument";

// Content Properties
export const USERNAME = "Username";
export const ID = "Id";

// Content Actions
export const DELETE = "Delete";
export const CLOSE = "Close";

// Sort Types
export const DESCENDING = "desc";
export const ASCENDING = "asc";
export const SORT_BY_VALUES = [DESCENDING, ASCENDING];

// Content Sort By Values
export const NEWEST = "Newest";
export const OLDEST = "Oldest";
export const MOST_DISCUSSED = "Most Discussed";
export const LEAST_DISCUSSED = "Least Discussed";
export const DISPLAY_SORT_BY_VALUES = [NEWEST, OLDEST, MOST_DISCUSSED, LEAST_DISCUSSED];

// Used in Homepage
export const NO_OF_MOST_DISCUSSED_PETITIONS = 3;
export const NO_OF_MOST_DISCUSSED_ARGUMENTS = 3;

// Petition Form Fields
export const PETITION_TITLE_ID = "title";
export const PETITION_SITUATION_ID = "situation";
export const PETITION_ACTION_ID = "action";
export const PETITION_GOAL_ID = "goal";
export const PETITION_VALUE_ID = "value";
export const PETITION_CLOSING_DATE_ID = "closingDateTime";
export const PETITION_IS_ANONYMOUS_ID = "isAnonymous";

// Register User Fields
export const USERNAME_ID = "username";
export const FIRST_NAME_ID = "firstName";
export const LAST_NAME_ID = "lastName";
export const PASSWORD_ID = "password";
export const CONFIRM_PASSWORD_ID = "confirm-password";

// Register User Field Validation
export const PASSWORD_LENGTH_MIN = 5;
export const PASSWORD_LENGTH_MAX = 21;
export const LAST_NAME_LENGTH_MAX = 40;
export const FIRST_NAME_LENGTH_MAX = 40;
export const USERNAME_LENGTH_MAX = 15;
export const MAXIMUM_NO_OF_SOURCES = 2;
export const MINIMUM_NO_OF_SOURCES = 0;

// Petition Form Validation
export const TITLE_LENGTH_MAX = 90;
export const PETITION_FIELD_MAX = 3000;

// Argument Form Validation
export const EXPERT_FIELD_MAX = 90;
export const ARGUMENT_FIELD_MAX = 6000;

// For Argument Data Visualisation
export const TREE_NETWORK_HEIGHT = 400;

// Other
export const VALID_URL_PREFIX_REGEX = /^(https:\/\/|http:\/\/).*/;
export const PETITION_LIST_SIZE = 5;
export const NOTIFICATION_AUTO_CLOSE_MS = 5000;
export const PETITION_ENTRY_SITUATION_MAX_LENGTH = 400;
export const PETITION_ENTRY_SITUATION_ALT_MAX_LENGTH = 200;
export const MOBILE_MAX_WIDTH = 768;