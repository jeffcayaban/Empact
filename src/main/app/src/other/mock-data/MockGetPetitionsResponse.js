export default {
    "content": [
        {
            "id": 1,
            "title": "TestTitle",
            "situation": "TestSituation",
            "action": "TestAction",
            "goal": "TestGoal",
            "value": "TestValue",
            "creationDateTime": "2019-02-14T13:04:29Z",
            "lastUpdatedDateTime": "2019-02-14T13:04:38Z",
            "createdBy": {
                "id": 1,
                "username": "admin",
                "firstName": "Admin",
                "lastName": "Account",
                "authorities": [
                    "ROLE_ADMIN"
                ]
            },
            "closingDateTime": "2019-02-15T13:04:27Z",
            "isAnonymous": false,
            "isWinning": true
        }
    ],
    "page": 0,
    "size": 5,
    "totalElements": 1,
    "totalPages": 1,
    "last": true
}