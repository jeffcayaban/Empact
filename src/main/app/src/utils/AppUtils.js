import {API_BASE_URL, AUTHORIZATION_HEADER_FIELD} from '../constants';
import {ACCESS_TOKEN, AUTHORIZATION_TOKEN_TYPE, PETITION_LIST_SIZE} from "../constants";

/** Creates a request object and sends the request to the server. **/
const request = (options) => {
    const headers = new Headers({ 'Content-Type': 'application/json' });
    const accessToken = localStorage.getItem(ACCESS_TOKEN);

    if (accessToken !== undefined && accessToken !== null) {
        headers.append(AUTHORIZATION_HEADER_FIELD, `${AUTHORIZATION_TOKEN_TYPE} ${accessToken}`)
    }

    options = {...options, headers: headers};

    return fetch(options.url, options).then(response =>
            (response.json()).then(jsonResponse => (!response.ok) ? Promise.reject(jsonResponse) : jsonResponse)
    );
};

// ---------------------------- APIs ----------------------------

/** Gets all the petitions **/
export function getAllPetitions(page, size, sort, showClosed) {
    page = page || 0;
    size = size || PETITION_LIST_SIZE;

    return request({
        url: API_BASE_URL + "/petitions?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Gets a single petition given an ID **/
export function getPetition(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/" + petitionId,
        method: 'GET'
    });
}

/** Creates a new petition given its details **/
export function createPetition(petitionData) {
    return request({
        url: API_BASE_URL + "/petitions/create",
        method: 'POST',
        body: JSON.stringify(petitionData)
    });
}

/** Deletes an existing petition given its details **/
export function deletePetition(deletePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/delete",
        method: 'DELETE',
        body: JSON.stringify(deletePetitionData)
    });
}

/** Updates an existing petition with the given details **/
export function editPetition(editPetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/edit/",
        method: 'POST',
        body: JSON.stringify(editPetitionData)
    });
}

/** Logs in into a user account **/
export function login(loginRequest) {
    return request({
        url: API_BASE_URL + "/auth/signInUser",
        method: 'POST',
        body: JSON.stringify(loginRequest)
    });
}

/** Creates a new user account given the account details **/
export function signup(signupRequest) {
    return request({
        url: API_BASE_URL + "/auth/signUpUser",
        method: 'POST',
        body: JSON.stringify(signupRequest)
    });
}

/** Checks the username's availability given a username **/
export function checkUsernameAvailability(username) {
    return request({
        url: API_BASE_URL + "/user/checkUsernameAvailability?username=" + username,
        method: 'GET'
    });
}

/** Deletes a user **/
export function deleteUser(deleteUserRequest) {
    return request({
        url: API_BASE_URL + "/auth/deleteUser",
        method: 'DELETE',
        body: JSON.stringify(deleteUserRequest)
    })
}

/** Fetches the current user **/
export function getCurrentUser() {
    if(!localStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }

    return request({
        url: API_BASE_URL + "/user/getCurrentUser",
        method: 'GET'
    });
}

/** Fetches a user profile given a user's username **/
export function getUserProfile(username) {
    return request({
        url: API_BASE_URL + "/user/getUserProfile?username=" + username,
        method: 'GET'
    });
}

/** Fetches the petitions for a given user **/
export function getUserPetitions(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/user?username=" + username + "&page=" + page + "&size=" + size + "&sort=" + sort
        + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Fetches the most discussed petitions for a given user **/
export function getMostDiscussedPetitionsByUser(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitionsByUser?username=" + username + "&page=" + page +
            "&size=" + size + "&sort=" + sort
            + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Closes a petition given its details **/
export function closePetition(closePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/close",
        method: 'POST',
        body: JSON.stringify(closePetitionData)
    })
}

/** Creates a new argument by expert opinion given the new details **/
export function createArgumentByExpertOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Creates a new argument by popular opinion given the new details **/
export function createArgumentByPopularOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Fetches the arguments for a given petition **/
export function getArgumentsByRootPetitionId(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByRootPetitionId?rootPetitionId=" + petitionId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the arguments for a given argument **/
export function getArgumentsByArgumentId(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByArgumentId?argumentId=" + argumentId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the argument for a given ID **/
export function getArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/" + argumentId,
        method: 'GET'
    });
}

/** Fetches the parent argument for a given argument **/
export function getParentArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/getParentArgumentById?argumentId=" + argumentId,
        method: 'GET',
    });
}

/** Checks whether a petition is closed or not **/
export function isPetitionClosed(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/isPetitionClosed/" + petitionId,
        method: 'GET'
    })
}

/** Fetches the arguments for a given user **/
export function getUserArguments(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByUsernameAndAnonymity?username=" + username +
            "&showAnonymousContent=" + showAnonymousContent + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET'
    })
}

/** Fetches the number of supporting or opposing arguments for a given petition **/
export function countArgsByPetitionIdAndIsSupporting(petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countArgsByPetitionIdAndIsSupporting?petitionId=" + petitionId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Fetches the number of supporting or opposing arguments for a given argument **/
export function countArgsByArgumentIdAndIsSupporting(argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countAllByArgumentIdAndIsSupporting?argumentId=" + argumentId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Deletes an argument given its details **/
export function deleteArgument(deleteArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/delete",
        method: 'DELETE',
        body: JSON.stringify(deleteArgumentData)
    });
}

/** Updates an existing argument by popular opinion with new details **/
export function editArgumentByPopularOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Updates an existing argument by expert opinion with new details **/
export function editArgumentByExpertOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Fetches the most discussed petitions **/
export function getMostDiscussedPetitions(page, size, sort, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitions?page=" + page + "&size=" + size + "&sort=" + sort +
            "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments **/
export function getMostDiscussedArguments(pageSize, showClosed) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArguments?size=" + pageSize + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given petition **/
export function getMostDiscussedArgumentsForPetition(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForPetition?page=" + page + "&size=" + size + "&sort=" + sort +
            "&petitionId=" + petitionId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given argument **/
export function getMostDiscussedArgumentsForArgument(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForArgument?page=" + page + "&size=" + size + "&sort=" + sort +
            "&argumentId=" + argumentId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given user **/
export function getMostDiscussedArgumentsForUser(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForUser?page=" + page + "&size=" + size + "&sort=" + sort +
            "&username=" + username + "&showAnonymousContent=" + showAnonymousContent,
        method: 'GET'
    });
}

/** Fetches the data nodes required for the data visualisation of arguments **/
export function getDataNodes(contentId, isPetition) {
    return request({
        url: API_BASE_URL + "/arguments/getDataNodes?contentId=" + contentId + "&isPetition=" + isPetition,
        method: 'GET'
    });
}

/** Fetches data required for the petition report **/
export function getPetitionReport(petitionId) {
    return request({
        url: API_BASE_URL + "/arguments/getPetitionReport?petitionId=" + petitionId,
        method: 'GET'
    });
}

/** Searches for petitions given a query **/
export function searchPetitions(page, size, sort, showClosed, query) {
    return request({
        url: API_BASE_URL + "/petitions/search?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed
         + "&query=" + query,
        method: 'GET'
    });
}
