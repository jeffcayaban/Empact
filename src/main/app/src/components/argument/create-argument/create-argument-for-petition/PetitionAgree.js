import React from 'react'
import {Redirect, Route} from "react-router-dom";
import { CURRENT_USERNAME } from "../../../../constants";
import {getPetition, isPetitionClosed} from "../../../../utils/AppUtils";
import Loader from "../../../Loader";
import SituationCQ1 from "./critical-questions/SituationCQ1";
import ActionCQ3 from "./critical-questions/ActionCQ3";
import {Button} from "react-bootstrap";
import NoArgCreationClosedPetition from "../../../common/alerts/NoArgCreationOnClosedPetitionAlert";
import ContentRetrievalAlert from "../../../common/alerts/ContentRetrievalAlert";
import {PETITION} from "../../../../constants";

/**
 * Will be displayed when the user decides to create an argument that agrees with a petition. It will display
 * relevant critical questions that the user can choose to answer.
 */

class PetitionAgree extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            isLoading: true,
            petition: null,
            isPetitionClosed: null,
        }
    }

    // Used for fetching the petition that is to be made to.
    getPetitionFromServer(petitionId) {
        getPetition(petitionId)
            .then(response => this.setState({ isLoading: false, petition: response }))
            .catch(error => this.setState({ isLoading: false }))
    }

    // Used for fetching whether the root petition is closed or not.
    getIsPetitionClosedFromServer(petitionId) {
        isPetitionClosed(petitionId)
            .then(response => this.setState({ isPetitionClosed: response }))
            .catch(error => console.log(error));
    }

    componentDidMount() {
        const petitionId = this.props.match.params.id;
        this.getPetitionFromServer(petitionId);
        this.getIsPetitionClosedFromServer(petitionId);
    }

    render() {
        const petitionId = this.props.match.params.id;
        const username = localStorage.getItem(CURRENT_USERNAME);
        const { isLoading, isPetitionClosed, petition } = this.state;

        if (!isLoading) {
            // Assumes that any data fetching has finished.

            if (petition !== null && petition !== undefined) {
                // Assumes that the petition has been fetched.

                if (username && !isPetitionClosed) {
                    // Assumes the user is logged in and the petition is not closed.

                    return (
                        <div className={"secondaryPageContainerStyle"}>
                            <div className={"centerStyle"}>
                                <h2 align="center">You <strong>agree</strong> with the petition</h2>
                                <p align="center">Tell us your reasons for deciding this!</p>
                            </div>

                            <div style={{ paddingTop: '30px' }}>
                                <SituationCQ1 petitionId={petitionId} isSupporting={true} situation={petition.situation} />
                                <ActionCQ3 petitionId={petitionId} isSupporting={true} petition={petition} />
                            </div>
                            <Route render={({ history }) => (
                                <Button onClick={() => { history.push(`/petition/${petitionId}`); }}>Back</Button>
                            )}/>
                        </div>
                    );
                } else if (isPetitionClosed) {
                    // Assumes the petition is closed.
                    return <NoArgCreationClosedPetition />
                } else {
                    // Assumes the user is not logged in and will be redirected to the login page.
                    return <Redirect to={{ pathname: '/login' }} />
                }
            } else {
                // Will display an alert when the petition could not be fetched.
                return <div className={"alertContainer"}><ContentRetrievalAlert contentType={PETITION} /></div>;
            }
        } else {
            // Will display a loader when data is being fetched.
            return (<Loader isLoading={isLoading} />);
        }
    }
}

export default PetitionAgree;