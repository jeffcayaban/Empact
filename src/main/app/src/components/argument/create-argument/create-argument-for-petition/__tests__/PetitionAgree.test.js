import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import PetitionAgree from "../PetitionAgree";
import * as AppUtils from "../../../../../utils/AppUtils";
import {CURRENT_USERNAME} from "../../../../../constants";
import MockGetPetitionsResponse from '../../../../../other/mock-data/MockGetArgumentsResponse';

describe('PetitionAgree', () => {

    it('should render a loader', () => {
        const component = shallow(
            <PetitionAgree match={{ params: { id: 'testId' } }} />
        );

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render a content retrieval alert', () => {
        const component = shallow(<PetitionAgree match={{ params: { id: 'testId' } }} />);
        component.setState({ petition: null, isPetitionClosed: false, isLoading: false });
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should successfully render the page given the petition is open', () => {
        const petitionData = MockGetPetitionsResponse.content[0];

        const originaGetPetition = AppUtils.getPetition;
        const originalIsPetitionClosed = AppUtils.isPetitionClosed;
        AppUtils.getPetition = () => Promise.resolve(petitionData);
        AppUtils.isPetitionClosed = () => Promise.resolve(false);

        localStorage.setItem(CURRENT_USERNAME, "test");

        const component = shallow(<PetitionAgree match={{ params: { id: 'testId' } }} />);
        component.setState({ petition: petitionData, isPetitionClosed: false, isLoading: false });

        expect(toJson(component)).toMatchSnapshot();

        localStorage.removeItem(CURRENT_USERNAME);

        AppUtils.getPetition = originaGetPetition;
        AppUtils.isPetitionClosed = originalIsPetitionClosed;
    });

    it('should successfully render the page given the petition is closed.', () => {
        const petitionData = MockGetPetitionsResponse.content[0];

        const originaGetPetition = AppUtils.getPetition;
        const originalIsPetitionClosed = AppUtils.isPetitionClosed;
        AppUtils.getPetition = () => Promise.resolve(petitionData);
        AppUtils.isPetitionClosed = () => Promise.resolve(false);

        localStorage.setItem(CURRENT_USERNAME, "test");

        const component = shallow(<PetitionAgree match={{ params: { id: 'testId' } }} />);
        component.setState({ petition: petitionData, isPetitionClosed: true, isLoading: false });

        expect(toJson(component)).toMatchSnapshot();

        localStorage.removeItem(CURRENT_USERNAME);

        AppUtils.getPetition = originaGetPetition;
        AppUtils.isPetitionClosed = originalIsPetitionClosed;
    });

});
