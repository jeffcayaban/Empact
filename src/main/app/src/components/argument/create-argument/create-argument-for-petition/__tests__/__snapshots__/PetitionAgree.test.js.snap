// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetitionAgree should render a content retrieval alert 1`] = `
<div
  className="alertContainer"
>
  <ContentRetrievalAlert
    contentType="Petition"
  />
</div>
`;

exports[`PetitionAgree should render a loader 1`] = `
<Loader
  isLoading={true}
/>
`;

exports[`PetitionAgree should successfully render the page given the petition is closed. 1`] = `<NoArgCreationOnClosedPetitionAlert />`;

exports[`PetitionAgree should successfully render the page given the petition is open 1`] = `
<div
  className="secondaryPageContainerStyle"
>
  <div
    className="centerStyle"
  >
    <h2
      align="center"
    >
      You 
      <strong>
        agree
      </strong>
       with the petition
    </h2>
    <p
      align="center"
    >
      Tell us your reasons for deciding this!
    </p>
  </div>
  <div
    style={
      Object {
        "paddingTop": "30px",
      }
    }
  >
    <SituationCQ1
      isSupporting={true}
      petitionId="testId"
    />
    <ActionCQ3
      isSupporting={true}
      petition={
        Object {
          "agreesWithCQ": true,
          "argumentType": "PopularOpinion",
          "createdBy": Object {
            "authorities": Array [
              "ROLE_ADMIN",
            ],
            "firstName": "Admin",
            "id": 1,
            "lastName": "Account",
            "username": "admin",
          },
          "creationDateTime": "2019-02-14T13:04:38Z",
          "criticalQuestionId": "CQ2",
          "explanation": "TestExplanation",
          "id": 1,
          "isAnonymous": false,
          "isSupporting": true,
          "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
          "parentArgumentId": null,
          "rootPetitionId": 1,
          "sources": Array [],
        }
      }
      petitionId="testId"
    />
  </div>
  <Route
    render={[Function]}
  />
</div>
`;
